libs/sarif/src/schema/run.rs

```diff
@@ -0,0 +1,46 @@
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use crate::schema::Tool;
+
+/// Describes a single run of an analysis tool, and contains the reported output of that run.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(
+    feature = "serde",
+    derive(Serialize, Deserialize),
+    serde(rename_all = "camelCase")
+)]
+#[non_exhaustive]
+pub struct Run {
```

Comment:
If `results` is present `columnKind` (`3.14.27`) must also be present

Reply:
This is also missing `redactionToken` (`3.14.28`) which is required if any property that is redactable (`3.5.2`) has been redacted.

This plays into account, especially with the `invocation` object, which has quite a lot of redactable properties.

Reply:
I have not implemented the `Run` members yet, this is subject to a follow-up

Reply:
Looking back into the spec the following is the case:

> If the tool started but failed to begin its analysis (for example, because its command line was invalid), then again results MAY be present, and if present SHALL be null.

> In all other circumstances, results SHALL be present and SHALL contain all results detected by the tool. If the tool did not detect any results, results SHALL be an empty array.

> If results is absent, it SHALL default to null.

This means that indeed results is optional, but this implies (due to the value being `null`) that the tool started but failed to begin it's analysis.